crates/ruma-api-macros/src/version.rs

```diff
@@ -0,0 +1,51 @@
+use std::{convert::TryInto, num::NonZeroU8, str::FromStr};
+
+use proc_macro2::TokenStream;
+use quote::ToTokens;
+use syn::{parse::Parse, Error, LitFloat};
+
+#[derive(Clone)]
+pub struct MatrixVersionLiteral {
+    maj: NonZeroU8,
+    min: u8,
+}
+
+impl Parse for MatrixVersionLiteral {
+    fn parse(input: syn::parse::ParseStream<'_>) -> syn::Result<Self> {
+        let fl: LitFloat = input.parse()?;
+
+        if !fl.suffix().is_empty() {
+            return Err(Error::new_spanned(
+                fl,
+                "matrix version variable contained invalid float suffix",
+            ));
+        }
+
+        let ver_vec: Vec<&str> = fl.base10_digits().split('.').collect();
```

Comment:
I wouldn't trust `base10_digits()` to give the original representation as found in the source. `to_string()` should do that though:

```suggestion
        let ver_vec: Vec<&str> = fl.to_string().split('.').collect();
```

Reply:
You'd be right, I had an interesting bug when I did not specify the 0 after the ., but which I didn't remember until you mentioned this error.